ds/dataset.go

```diff
@@ -44,26 +44,34 @@ type MutateFieldCheck func(path ...string) error
 
 // Dataset is a qri dataset starlark type
 type Dataset struct {
-	ds    *dataset.Dataset
-	body  starlark.Iterable
-	check MutateFieldCheck
+	read    *dataset.Dataset
+	write   *dataset.Dataset
+	body    starlark.Iterable
+	check   MutateFieldCheck
+	modBody bool
 }
 
 // NewDataset creates a dataset object, intended to be called from go-land to prepare datasets
 // for handing to other functions
 func NewDataset(ds *dataset.Dataset, check MutateFieldCheck) *Dataset {
-	return &Dataset{ds: ds, check: check}
+	return &Dataset{read: ds, check: check}
 }
 
 // New creates a new dataset from starlark land
 func New(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	d := &Dataset{ds: &dataset.Dataset{}}
+	d := &Dataset{read: &dataset.Dataset{}, write: &dataset.Dataset{}}
 	return d.Methods(), nil
 }
 
 // Dataset returns the underlying dataset
 func (d *Dataset) Dataset() *dataset.Dataset {
-	return d.ds
+	// TODO(dlong): What to do here?
```

Comment:
Did a quick audit, I think we can remove this method entirely

Reply:
Excellent, that was the outcome I was hoping for.